bindings/py/cpp_src/bindings/algorithms/py_SpatialPooler.cpp

```diff
@@ -180,8 +186,11 @@ namespace nupic_ext
             {
                 throw runtime_error("Incompatible format. Expect uint32");
             }
-
-            self.compute(get_it<UInt>(x), learn, get_it<UInt>(y));
+            SDR in( self.getInputDimensions() );
+	    SDR out(self.getColumnDimensions() );
+	    in.setDense(get_it<UInt>(x));
+            self.compute(in, learn, out);
+	    y = out.getSparse(); //FIXME how pass SDR to array? 
```

Comment:
@dkeeney what would be the correct way to cast (c++)SDR to numpy array? 
I could use py::SDR, but I'd like to keep the interface, and think np::array is more useful there.

Reply:
It would be nice if a .py user could call it either way.  numpy is flexible but slow. For new .py apps they could use the py::SDR but the old stuff will need the numpy interface.

The numpy interface is a problem...do you give them the sparse array or the dense array?  If I remember correctly, I think it is usually expecting dense arrays.  numpy gives you a buffer and you copy the result from your SDR.   Don't know if that helps.

Reply:
> The numpy interface is a problem.

I have similar problem in #311 , if we used `[]` list, pybind would be able to convert automatically. 

> .do you give them the sparse array or the dense array? 

I have an SDR, so both (sparse, binary) at hand. So I'd like to keep this consistent with what other algorithms give. 

>  numpy gives you a buffer and you copy the result from your SDR. Don't know if that helps.

I'll try that. So resize np array to fit the SDR, and copy to its buffer (?) 
Thank you

Reply:
I think that the python should only use the new SDR API.  If the user wants numpy then they should access the property / attribute `sdr.dense` which returns a numpy array, like the old API did.

Reply:
I like that.  
I know we don't want to put logic in the .py bindings code but that could be hid in the bindings so legacy .py code that is using numpy would not be aware that SDR's are being passed on into C++ and returned on its behalf.

Reply:
> f the user wants numpy then they should access the property / attribute sdr.dense

I agree this would be a solution. But as I was argumenting for the breaking change in C++ (according to survey), here in py you'd break the python API. Is that ok? 
On the other hand, we removed SparseMatrix, so we broke some of it anyway. 

>  so legacy .py code that is using numpy would not be aware that SDR's are being passed on into C++ and returned on its behalf. 

this is not about what we call from py->c++ (there we'll use the SDR variant), but from c++ I get a SDR. The question is what do I return? nparray / sdr? 
(this discussion started as request for help how to convert a sdr to np array)